heartbeat/gcp-ilb

```diff
@@ -0,0 +1,338 @@
+    </parameter>
+
+</parameters>
+
+<actions>
+    <action name="start"        timeout="10s" />
+    <action name="stop"         timeout="30s" />
+    <action name="monitor"      depth="0" timeout="30s" interval="30s" />
+    <action name="validate-all" timeout="5s" />
+    <action name="meta-data"    timeout="5s" />
+</actions>
+</resource-agent>
+END
+exit 0
+}
+
+#######################################################################
+
+log() {
+	lvl=$1
+	msg=$2
+	ocf_log $lvl "$0 - $process - $pid: $msg"
+	if [[ "${OCF_RESKEY_log_enable}" = "true" ]] ; then
+	    ( ${OCF_RESKEY_log_cmd} ${OCF_RESKEY_log_params} "L $lvl: $msg" ${OCF_RESKEY_log_end_params} )
+	fi
+}
+
+getpid() {
+	if ! [[ -f "$pidfile" ]] ; then
+		return 
+	fi
+	
+	cat $pidfile
+}
+
+ilb_usage() {
+    cat <<END
+usage: $0 {start|stop|monitor|status|meta-data|validate-all}
+
+Agent wrapping socat or nc to reply to health probes.
+END
+}
+
+
+
+
+ilb_monitor() {
+	
+	pid=`getpid`
+	log debug "pid is $pid"
+	
+	if  [[ -z "$pid" ]] ; then
+		return $OCF_NOT_RUNNING
+	fi
+	
+	if [[ -n "$pid" ]] && kill -s 0 $pid ; then
+        log debug "Process is currently running"
+		return $OCF_SUCCESS
+	else 
+		log warn "The process is not running but has a pidfile. Removing file"
+		rm -f $pidfile
+		return $OCF_NOT_RUNNING
+	fi
+
+}
+
+ilb_start() {
+	
+	if ilb_monitor; then
+		log debug "Process is already running"
+		return $OCF_SUCCESS
+	fi
+	
+	cmd="$OCF_RESKEY_cat -U TCP-LISTEN:$OCF_RESKEY_port,backlog=10,fork,reuseaddr /dev/null"
+
+	if [ $( basename $OCF_RESKEY_cat ) = 'nc' ]; then
+		cmd="$OCF_RESKEY_cat -l -k $OCF_RESKEY_port"
+	fi
+	
+	log debug "Starting with \'$cmd\'"
+	( ${cmd}  ) &   pid="$!" 
+	disown
+
+	if [[ -n "$pid" ]] ; then
+		log debug "$pid is started"
+		echo "$pid" > $pidfile
+		return $OCF_SUCCESS
+	else
+		log err "\'$cmd\' could not be started"
+		return $OCF_ERR_GENERIC
+	fi
+
+} 
+
+ilb_stop() {
+	
+	if ! ilb_monitor; then
+		rm -f $pidfile
+		return $OCF_SUCCESS
+	fi
+
+	if [[ -n "$OCF_RESKEY_CRM_meta_timeout" ]]; then
+		# Allow 2/3 of the action timeout for the orderly shutdown
+		# (The origin unit is ms, hence the conversion)
+		stop_timeout=$((OCF_RESKEY_CRM_meta_timeout/1500))
+	else
+		stop_timeout=10
+	fi
+
+
+	pid=`getpid`
+	kill $pid
+
+	i=0
+	while [ $i -lt $stop_timeout ]; do
+		if ! ilb_monitor; then
+			rm -f $pidfile
+			return $OCF_SUCCESS
+		fi
+		sleep 1 
+		i=$((i+1))
+	done
+
+	log warn "Stop with SIGTERM failed/timed out, now sending SIGKILL."
+
+	i=0
+	while [ $i -lt $stop_timeout ]; do
+
+		kill -s 9 $pid
+
+		if ! ilb_monitor; then
+			log warn "SIGKILL did the job."
+			rm -f $pidfile
+			return $OCF_SUCCESS
+		fi
+		log info "The job still hasn't stopped yet. Re-trying SIGKILL..."
+		sleep 2
+		i=$((i+2))
+	done
+
+	log err "The cat has more than 9 lives and could not be terminated."
+	return $OCF_ERR_GENERIC
+	
+}
+
+ilb_validate() {
+	check_binary "$OCF_RESKEY_cat"
+    check_binary "$OCF_RESKEY_log_cmd"
```

Comment:
Update to tab here to follow the indentation of the rest of the code.

Reply:
Done and committed fix. Thanks for reviewing